maxim_grabovskiy/main.cpp

```diff
@@ -18,33 +19,49 @@ class Graph {
     EdgeId const id;
     VertexId const fromVertexId;
     VertexId const toVertexId;
-    Edge(EdgeId _id, VertexId _fromVertexId, VertexId _toVertexId)
-        : id(_id), fromVertexId(_fromVertexId), toVertexId(_toVertexId) {}
+    string const color;
+    Edge(EdgeId _id,
+         VertexId _fromVertexId,
+         VertexId _toVertexId,
+         string _color)
+        : id(_id),
+          fromVertexId(_fromVertexId),
+          toVertexId(_toVertexId),
+          color(_color) {}
   };
   struct Vertex {
     VertexId const id;
-    explicit Vertex(VertexId _id) : id(_id) {}
+    int depth;
+    explicit Vertex(VertexId _id, int _depth) : id(_id), depth(_depth) {}
   };
   vector<EdgeId> const& getConnectionsVector(VertexId vertexId) const {
     return edgeConectionMap_.at(vertexId);
   }
+  vector<EdgeId> const& getVertexIdByDepth(int depth) const {
+    return vertexIdByDepth_.at(depth);
+  }
   vector<Vertex> const& getVertexes() const { return vertexes_; }
   vector<Edge> const& getEdges() const { return edges_; }
-  void addEdge(VertexId fromVertexId, VertexId toVertexId) {
+
+  void addEdge(VertexId fromVertexId, VertexId toVertexId, string color) {
     EdgeId const newEdgeId = getNewEdgeId();
     edgeConectionMap_[fromVertexId].push_back(newEdgeId);
     edgeConectionMap_[toVertexId].push_back(newEdgeId);
-    edges_.emplace_back(newEdgeId, fromVertexId, toVertexId);
+    edges_.emplace_back(newEdgeId, fromVertexId, toVertexId, color);
   }
   VertexId addVertex() {
     VertexId const newVertexId = getNewVertexId();
-    vertexes_.emplace_back(newVertexId);
+    vertexes_.emplace_back(newVertexId, 0);
+    vertexIdByDepth_[0].emplace_back(0);
```

Comment:
Нельзя брать данные с потолка
```suggestion
    const auto& new_vertex = vertexes_.emplace_back(newVertexId, 0);
    vertexIdByDepth_[0].emplace_back(new_vertex.id);
```